### PR TITLE
[WIP] Update CRD for client validation

### PR DESCRIPTION
**Pull request by carmark**

### CODE REVIEW

File: deploy/mpi-operator.yaml [hunk 1]
```diff
@@ -1,51 +1,11 @@
 # --------------------------------------------------
-# - Single configuration deployment YAML for MPI-Operator
+# - Configuration deployment YAML for MPI-Operator
```
Comment (terrytangyuan) [inline deploy/mpi-operator.yaml:2]: This file was originally purposed to provide one single file for MPI Operator installation. Maybe have a script to merge the generated CRD yaml with this one?

File: deploy/mpijob-v1alpha2-crd.yaml [hunk 2]
```diff
@@ -0,0 +1,5644 @@
+# --------------------------------------------------
+# - Configuration deployment YAML for MPI-Operator
+# - Include:
+#      CRD deployment 
+# --------------------------------------------------
+# The code is generated by crd-validation
```
Comment (terrytangyuan) [inline deploy/mpijob-v1alpha2-crd.yaml:6]: Are you using this? https://github.com/kubeflow/crd-validation
Does k8s provide other official solutions?

File: deploy/mpijob-v1alpha2-crd.yaml [hunk 3]
```diff
@@ -0,0 +1,5644 @@
...
+---
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (terrytangyuan) [inline deploy/mpijob-v1alpha2-crd.yaml:8]: Could you add a script or instruction on how to autogen this?

Comment (terrytangyuan) [reply deploy/mpijob-v1alpha2-crd.yaml:8]: We probably need to update this guide as well (e.g. using Makefile, etc.): https://github.com/kubeflow/mpi-operator/blob/master/CONTRIBUTING.md

**Comment by terrytangyuan** [comment]
Any update on this? This is related to "CRD schema validation" requirement outlined in https://github.com/kubeflow/mpi-operator/issues/217.

File: deploy/mpijob-v1alpha2-crd.yaml (hunk 2 above)
Comment (carmark) [reply deploy/mpijob-v1alpha2-crd.yaml:6]: yes, this validation is generated by [crd-validation](https://github.com/kubeflow/crd-validation).  This openAPIV3Schema should be the official solution to validate the CRD which include pod template.

File: deploy/mpijob-v1alpha2-crd.yaml (hunk 3 above)
Comment (carmark) [reply deploy/mpijob-v1alpha2-crd.yaml:8]: yes, I will update the doc and script to autogen it.

**Comment by carmark** [comment]
@terrytangyuan The mpijob validation in [crd-validation](https://github.com/kubeflow/crd-validation) is waiting for merging, and will update related doc after that.

**Comment by terrytangyuan** [comment]
For the records, here's the PR to support MPIJob CRD validation https://github.com/kubeflow/crd-validation/pull/9

File: deploy/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# How to create a validation CRD for mpijob
+
+* Submit a PR to update the vendor version for [crd-validation](https://github.com/github.com/crd-validation)
+* Build and generate a new API validation CRD
+  * build the [crd-validation](https://github.com/github.com/crd-validation)
```
Comment (terrytangyuan) [inline deploy/README.md:5]: wrong link?

File: deploy/README.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+  * `./crd-validation -c ./examples/mpijob/crd-validation.yaml mpijob`
+* Copy it to this directory
```
Comment (terrytangyuan) [inline deploy/README.md:7]: Perhaps add this as a script in /hack or Makefile? We also need to include a git hash for the crd-validation that we used

Comment (carmark) [reply deploy/README.md:7]: sure.

**Comment by terrytangyuan** [comment]
Closing due to inactivity.

**Pull request closed without merging**